tsl/src/bgw_policy/job.c

```diff
@@ -320,46 +283,31 @@ policy_retention_execute(int32 job_id, Jsonb *config)
 }
 
 bool
-policy_continuous_aggregate_execute(int32 job_id, Jsonb *config)
+policy_refresh_cagg_execute(int32 job_id, Jsonb *config)
 {
-	bool committed = false;
+	bool start_isnull, end_isnull;
 	int32 materialization_id;
-	bool finshed_all_materialization;
-	ContinuousAggMatOptions mat_options;
+	Hypertable *mat_ht;
 
-	materialization_id = policy_continuous_aggregate_get_mat_hypertable_id(config);
-
-	/* always materialize verbosely for now */
-	mat_options = (ContinuousAggMatOptions){
-		.verbose = true,
-		.within_single_transaction = false,
-		.process_only_invalidation = false,
-		.invalidate_prior_to_time = PG_INT64_MAX,
-	};
-
-	/* materialization code has it's own transaction handling so we commit
-	 * current transaction */
-	if (IsTransactionOrTransactionBlock())
-	{
-		committed = true;
-		if (ActiveSnapshotSet())
-			PopActiveSnapshot();
-		CommitTransactionCommand();
-	}
-
-	finshed_all_materialization = continuous_agg_materialize(materialization_id, &mat_options);
-
-	if (!finshed_all_materialization)
+	if (!ActiveSnapshotSet())
 	{
-		StartTransactionCommand();
-		enable_fast_restart(job_id, "materialize continuous aggregate");
-		CommitTransactionCommand();
+		PushActiveSnapshot(GetTransactionSnapshot());
 	}
-
-	/* start new transaction if we committed before */
-	if (committed)
-		StartTransactionCommand();
-
+	materialization_id = policy_continuous_aggregate_get_mat_hypertable_id(config);
+	mat_ht = ts_hypertable_get_by_id(materialization_id);
+	Dimension *open_dim = get_open_dimension_for_hypertable(mat_ht);
+	Oid dim_type = ts_dimension_get_partition_type(open_dim);
+	int64 refresh_start = policy_refresh_cagg_get_refresh_start(open_dim, config, &start_isnull);
+	int64 refresh_end = policy_refresh_cagg_get_refresh_end(open_dim, config, &end_isnull);
+	ContinuousAgg *cagg = ts_continuous_agg_find_by_mat_hypertable_id(materialization_id);
+	InternalTimeRange refresh_window = { .type = dim_type,
+										 .start = (start_isnull ? PG_INT64_MIN : refresh_start),
+										 .end = (end_isnull ? PG_INT64_MAX : refresh_end) };
+	cagg_refresh_internal(cagg, &refresh_window);
+	elog(LOG,
+		 "refresh cagg range %s , %s",
```

Comment:
Please spell out the complete name for the log.
```suggestion
		 "refresh continuous aggregate range %s , %s",
```

Reply:
fixed.